File: solutions/typescript/resistor-color/1/resistor-color.ts
export const colorCode = (color : String) : number => {
  const index = COLORS.indexOf(color.toLowerCase());

  if(index !== -1) return index;
  throw new Error(`Unknown color: ${color}`)
}

export const COLORS : string[] = [
        "black", "brown", "red", "orange", "yellow", 
        "green", "blue", "violet", "grey", "white"
    ];

File: solutions/java/resistor-color/1/src/main/java/ResistorColor.java
class ResistorColor {
    

    
    int colorCode(String color) {
       color = color.toLowerCase();
    for(int i = 0; i<10; i++){
        if(colors()[i].equals(color)){
            return i;
        }
    }
 throw new IllegalArgumentException("Invalid color: " + color);
    }

    String[] colors() {
        return new String[] {
        "black", "brown", "red", "orange", "yellow", 
        "green", "blue", "violet", "grey", "white"
    };
    }
}
